sched/distributor/poolDistributor.go

```diff
@@ -0,0 +1,63 @@
+package distributor
+
+import (
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+)
+
+//
+// Pool Distributor will assign work to any
+// Unscheduled Node in the cluster.
+//
+// While a Node is doing work no new tasks will
+// be sent to it.  Blocks until a node becomes free
+//
+// Static Cluster for now
+//
+type PoolDistributor struct {
+	freeCh chan cm.Node
+}
+
+//
+// Creates a Pool Distributor, assumes
+// all nodes in the cluster are free upon creation
+//
+// If the cluster has 0 nodes in it returns nil
+//
+func NewPoolDistributor(cluster cm.Cluster) *PoolDistributor {
+
+	nodes := cluster.Members()
+
+	if len(nodes) <= 0 {
+		return nil
+	}
+
+	// allocate the initial cluster size for the free channel
+	// assuming static clusters next update will support
+	// dynamic clusters
+	freeCh := make(chan cm.Node, len(nodes))
+
+	// put all the nodes in the freeCh
+	for _, n := range nodes {
+		freeCh <- n
+	}
+
+	return &PoolDistributor{
+		freeCh: freeCh,
+	}
+}
+
+//
+// Retrieve a Node to Do Work for this Particular Job
+//
+func (dist *PoolDistributor) ReserveNode(work sched.Job) cm.Node {
```

Comment:
Yes to some extent we will always need to say this node is no longer doing work.  Even if there are multiple tasks allowed to be scheduled on a node.  This code will get smarter over time, and I don't have the interface any more because this may evolve slightly and is pretty specific to individual schedulers.  I don't feel confident that I have the right abstraction yet